pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java

```diff
@@ -54,23 +54,23 @@ public RelDataType createRelDataTypeFromSchema(Schema schema) {
   private RelDataType toRelDataType(FieldSpec fieldSpec) {
     switch (fieldSpec.getDataType()) {
       case INT:
-        return createSqlType(SqlTypeName.INTEGER);
+        return createTypeWithNullability(createSqlType(SqlTypeName.INTEGER), true);
```

Comment:
Can you check this patch with the COLOCATED_JOIN quickstart and this query? I saw a broadcast distributed stage created for some reason. Also the projection wasn't pushed down to the userAttributes side.

```
SELECT
  COUNT(*)
FROM
  userAttributes_OFFLINE
WHERE
  userUUID NOT IN (
	SELECT userUUID FROM userGroups_OFFLINE WHERE groupUUID = 'group-2'
  )
```